common/src/prover_state/persistence.rs

```diff
@@ -26,21 +27,19 @@ fn get_serializers() -> (DefaultGateSerializer, DefaultGeneratorSerializer<Confi
 }
 
 #[inline]
-fn disk_path(circuit_config: &CircuitConfig) -> String {
-    format!(
-        "{}_{}",
-        PROVER_STATE_FILE_PREFIX,
-        circuit_config.get_configuration_digest()
-    )
+fn disk_path(circuit_config: &CircuitConfig, prefix: &str) -> String {
+    format!("{}_{}", prefix, circuit_config.get_configuration_digest())
 }
 
-pub fn from_disk(circuit_config: &CircuitConfig) -> Option<AllRecursiveCircuits> {
-    let path = disk_path(circuit_config);
+/// Loads [`AllRecursiveCircuits`] from disk.
+pub fn prover_from_disk(circuit_config: &CircuitConfig) -> Option<AllRecursiveCircuits> {
+    let path = disk_path(circuit_config, PROVER_STATE_FILE_PREFIX);
     let bytes = fs::read(&path).ok()?;
     info!("found prover state at {path}");
     let (gate_serializer, witness_serializer) = get_serializers();
     info!("deserializing prover state...");
-    let state = AllRecursiveCircuits::from_bytes(&bytes, &gate_serializer, &witness_serializer);
+    let state =
+        AllRecursiveCircuits::from_bytes(&bytes, false, &gate_serializer, &witness_serializer);
```

Comment:
This change and below is unrelated to the `VerifierState` change, but coming from the plonky2 dependency bump, necessary for splitting the transaction proof process and allowing workers to not load the entire set of table circuits from disk. I for now set the default to `false`, i.e. loading from and storing to disk the entire set of preprocessed circuits.